_posts/where-we-are-going.md

```diff
@@ -0,0 +1,80 @@
+
+For high-speed trains, tickets must come with "address" details to allow passengers to identify the reserved car and seat. Not to mention the reservation date and time.
+
+If seats are (micro)services, tickets are the service discovery infrastructure. They allow payloads to identify which specific service they need to fulfill the business requirement.
+
+If the system we're designing requires addressing instances, it suffers from spatial coupling.
+
+## Spatial coupling
+
+Spatial coupling is a form of coupling that surfaces in a design when two or more components, supposed to be autonomous, need to know "where" they are to operate. For example, connection strings to database servers are a form of spatial coupling. [SQL Server offers aliases](https://docs.microsoft.com/en-us/sql/tools/configuration-manager/aliases-sql-server-configuration-manager?view=sql-server-ver15). In a network of computers [DNS servers A and CNAME records](https://www.namecheap.com/guru-guides/dns-records/) masquerade the machines' physical addresses.
+
+If we take a closer look, they are all a form of service discovery. We need to connect to a database server or request data from a remote HTTP API. We depend on an external service, the alias mechanism, or the DNS server to reach the designated resource.
+
+## Why are balancers not enough?
+
+One could argue that a balancer in front of resources may solve the generated spatial coupling.
+
+When it comes to system resources, we might not need to connect to a specific node known in advance. We need one of the nodes. For example, if the requirement is to generate an invoice and the system offers a cluster of nodes that can create invoices, we need one, not a specific one. The "invoice creators" cluster could be proxied by a balancer that presents it as a single node and route requests internally.
+
+Still, we need to address the balancer. And we need service discovery.
+
+More importantly, we cannot group behind the same balancer different logical nodes. There cannot be an "invoice creator" node and an "order manager" node, both proxied by the same balancer. Well...we could, but then we also need a way to tell the balancer how to route requests behind the scenes.
+
+We're moving the cheese around. Spatial coupling changed its shape, but spatial coupling remains.
+
+If we wanted to use the train seat analogy, it would be like buying first-class or second-class tickets with no assigned seat. The train car is the balancer. Passengers will be required to find first or second-class cars and then pick a random free seat. The train coach is the load balancer. We cannot mix first and second-class tickets in the same carriage. It'll be a mess.
+
+## Back to the drawing board
+
+So far, we have learned a couple of things:
+
+- Spatial coupling is a problem we need to keep under control.

```

Comment:
We've identified that spatial coupling exists but I'm not sure we've identified it as a problem we need to keep under control yet

Reply:
Thanks, added a paragraph above.